### PR TITLE
Fix doc headers

### LINKED ISSUE #1874: Detect documentation bugs
## Issue Details

SInce the documentation is not inline for all packages, we should guarantee that documented header and real headers are the same. For example, we document `#include <CGAL/IO/output_surface_facets_to_off.h>` for `CGAL::output_surface_facets_to_off ()` while this header does not exist  (the real header is `CGAL/IO/Complex_2_in_triangulation_3_file_writer.h`)

- [ ] The simple thing we can do it to update `test_merge_of_branches` to check that any header in `doc/PKG/CGAL` exists. There might be false positive if the header is not in the package but I'm pretty sure we can workaround these.
- [ ] The second part of this issue is to fix the errors

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
This PR adds a script that checks if all the headers in Pkg/doc/pkg/CGAL actually have an equivalent. It also adds the test_merge_branch to the travis tests.
## Release Management

* Affected package(s): Scripts
* Issue(s) solved (if any): fix #1874
* Feature/Small Feature (if any):

### CODE REVIEW

**Comment by maxGimeno** [comment]
@lrineau  The script detected a license error in master

**Comment by sloriot** [comment]
`test_merge_of_branches` seems to be called twice for each job. Can't we simply call it once?

**Comment by maxGimeno** [comment]
I don't think I can do better than this, AFAIK, all the configuration is called for each job.

File: .travis.yml [hunk 1]
```diff
@@ -45,11 +45,13 @@ env:
   - PACKAGE='CHECK' 
 install: 
 before_script: 
+- sudo apt-get install zsh -y
```
Comment (lrineau) [inline .travis.yml]: No. Add `zsh` in the list `addons.apt.packages` instead. That avoids that apt is called twice.

File: .travis.yml [hunk 2]
```diff
@@ -45,11 +45,13 @@ env:
...
 - mkdir -p build 
 - cd build 
-- cmake -DCGAL_HEADER_ONLY=ON -DCMAKE_CXX_FLAGS_RELEASE=-DCGAL_NDEBUG .. 
+- cmake -DCGAL_HEADER_ONLY=ON -DWITH_demos:BOOL=TRUE -DWITH_examples:BOOL=true -DWITH_tests:BOOL=TRUE -DCMAKE_CXX_FLAGS_RELEASE=-DCGAL_NDEBUG .. 
 - make 
 - cd .. 
+- zsh -x -e ./Scripts/developer_scripts/test_merge_of_branch HEAD
```
Comment (lrineau) [inline .travis.yml]: That line `zsh -x -e ./Scripts/developer_scripts/test_merge_of_branch HEAD` should be in the `CHECK` section of `.travis/build_package.sh`.

Comment (lrineau) [inline .travis.yml]: By the way, please put the `PACKAGE='CHECK'` as the first job, to allow early detections of issues.

**Comment by lrineau** [comment]
For your information, the call to `test_merge_branch` in `.travis.yml` works, and has already detected one real issue in a PR of mines:
```
.. Checking include directives in header files...

The following files do not have an include directive of the package license header:

Mesh_3/include/CGAL/Polyhedral_complex_mesh_domain_3.h

(aborting the merge now)
```
https://travis-ci.org/CGAL/cgal/jobs/197708820

Thanks for this PR. That will help me a lot as a release manager!

**Pull request merged**